### ISSUE TITLE
git-lfs complains of missing file that shouldn't exist

### ISSUE THREAD

**Issue by qrkourier** [keybase/client#22502]
```bash
❯ git remote show keybase
Initializing Keybase... done.
Syncing with Keybase... done.
* remote keybase
  Fetch URL: keybase://private/kourier/nfwork
  Push  URL: keybase://private/kourier/nfwork
  HEAD branch: (unknown)

❯ git push keybase master                                                                                                                                                    
Initializing Keybase... done.                                                                                                                                                                 
Syncing with Keybase... done.                                                                                                                                                                 
LFS upload failed:cts:  94% (58/62), 1.1 GB | 10 MB/s                                          
  (missing) etc/edgePackerBuilder.gpg (f9bb35abbbd589d3e4a6893719fd8d783e16d3918f1d48648bfeb493f9d4410b)                                                                                      
  (missing) etc/artifactory.gpg (5c9562de768ad695d1bfe9b00931d2f8461a11b14ef1ddc81d9109a808afb26a)
  (missing) etc/credentials-digitalocean.gpg (35962f6f3f763444d52b195d7920f8a2bb0c2ca6f6e9976bc26490011e1b4662)                                                                               
  (missing) etc/sonarqube.gpg (c091081b03eaa09dde44ae069ec66c5f0dacd50731360ded1916964587241f50)
hint: Your push was rejected due to missing or corrupt local objects.                          
hint: You can disable this check with: 'git config lfs.allowincompletepush true'                                                                                                              
Uploading LFS objects:  94% (58/62), 1.1 GB | 10 MB/s, done                                                                                                                                   
error: failed to push some refs to 'keybase://private/kourier/nfwork'                          
                                                                                               
❯ for i in etc/{edgePackerBuilder,artifactory,credentials-digitalocean,sonarqube}.gpg; do file $i;done                                                                                        
etc/edgePackerBuilder.gpg: cannot open `etc/edgePackerBuilder.gpg' (No such file or directory) 
etc/artifactory.gpg: PGP RSA encrypted session key - keyid: E965E5F3 12D56B44 RSA (Encrypt or Sign) 4096b .
etc/credentials-digitalocean.gpg: PGP RSA encrypted session key - keyid: E965E5F3 12D56B44 RSA (Encrypt or Sign) 4096b .                                                                      
etc/sonarqube.gpg: PGP RSA encrypted session key - keyid: E965E5F3 12D56B44 RSA (Encrypt or Sign) 4096b .

❯ git --no-pager log --pretty=format:"[%h] %ae, %ar: %s" --stat -- etc/edgePackerBuilder.gpg etc/azure-credentials-edge-packer-builder.gpg                                                <<< 
[6815838] kenneth.bingham@netfoundry.io, 6 months ago: register changes
 etc/azure-credentials-edge-packer-builder.gpg | 3 +++
 etc/edgePackerBuilder.gpg                     | 3 ---
 2 files changed, 3 insertions(+), 3 deletions(-)

[5c17c7b] kenneth.bingham@netfoundry.io, 7 months ago: register changes
 etc/edgePackerBuilder.gpg | Bin 788 -> 128 bytes
 1 file changed, 0 insertions(+), 0 deletions(-)

[5f16dd6] kenneth.bingham@netfoundry.io, 11 months ago: register changes
 etc/edgePackerBuilder.gpg | Bin 0 -> 788 bytes
 1 file changed, 0 insertions(+), 0 deletions(-)

```

The missing file was moved to a different path some time ago, and now LFS is complaining that it's missing. I've deleted and recreated the keybase repo and re-ran `keybase git lfs-config`
my log id: 8f4ff98209fa6cb819e8171c

**Comment by strib** [comment]
To be clear, you moved the _local_ copies of your `etc/` files to a different path?  If so, and you haven't updated those paths in the repo itself, it seems like that is the issue, right?  I'm not an LFS expert, but I think you're not supposed to move the files without updating the git repo at the same time.  But I might be misunderstanding the issue here, please let me know.

**Comment by qrkourier** [comment]
The tracked file that's reported missing by LFS was `git mv` to current path in the shown commit from six months ago.

**Comment by strib** [comment]
Oh hmm.  KBFS didn't support LFS six months ago though, so I'm not sure how this could be something wrong with the KBFS repo (especially since you deleted and re-created the repo).  How did you port the LFS files over to KBFS initially?

Have you tried testing this repo out against different LFS providers?  Or cloning it from a different device?

(Also, you could try the config suggestion in the command output, but I'm guessing you'd prefer not to do that.)

**Comment by qrkourier** [comment]
> How did you port the LFS files over to KBFS initially?
After re-creating the private Keybase repo named "nfwork" I did `git remote add ...` and `keybase git lfs-config`

> Have you tried testing this repo out against different LFS providers? Or cloning it from a different device?
I was able to put the tracked files in .gitattributes to Bitbucket LFS.

> (Also, you could try the config suggestion in the command output, but I'm guessing you'd prefer not to do that.)
That's `git config lfs.allowincompletepush true` for our reference. I'm open to the idea, but didn't want to run it before I fully understood it.

Update: I've restored the file that was reported missing by LFS that had be previously `git mv` to a new path. I is no longer reported missing when I do `git push keybase`. Then, I did `git rm --cached` on the other files that are still reported missing despite being present in the location that's reported missing by LFS. Those three remaining files are still reported missing by LFS despite no longer being tracked by Git. The error still presents only when `git push keybase`, not `git push bitbucket`.

**Comment by qrkourier** [comment]
I did also `git add` those files and re-tried the push with the same outcome.

**Comment by qrkourier** [comment]
[Here's an ascii cast of re-creating the repo](https://asciinema.org/a/XpcRVNbmqZ4TDny7xWrfj4gr4) with `keybase` CLI and the failed push.

**Comment by strib** [comment]
My best guess is that your local checkout of the repo doesn't have the _historical_ versions of those old, now-moved LFS files, similar to what's described here: https://github.com/git-lfs/git-lfs/issues/2446

It doesn't error with bitbucket because bitbucket probably already has them, and they don't need to be uploaded.

If you can find a checkout somewhere that has the full history of the repo, it might work from there.  Otherwise, the linked issue seems to suggest turning on the `allowincompletepush` flag mentioned above.

Hope this helps!

**Comment by qrkourier** [comment]
From that linked issue I learned this problem is caused by using LFS with multiple remotes that are not mirrors i.e. have differences. If I'd created both remotes at the same time and kept them in sync the issue probably wouldn't have presented, but because I ran into [an issue with the keybase repo](https://github.com/keybase/client/issues/17804) at some point and stopped pushing there for a while it was missing some of the old LFS objects that had since been deleted.

I worked around this by setting `allowincompletepush 1`.

Thank you for your attention!
